test/h2spec/client.go

```diff
@@ -148,6 +150,7 @@ func run(cmd *cobra.Command, args []string) error {
 
 	success, err := h2spec.Run(c)
 	if !success {
+		log.Error(err, "error")
```

Comment:
The only error from `h2spec.Run`comes from the attempt to generate a junit report, so this is probably not useful, but otherwise lgtm.  https://github.com/openshift/cluster-ingress-operator/blob/8d9b42212a8ed0484f51d5a3cf9ede958878ad48/vendor/github.com/summerwind/h2spec/h2spec.go#L62

Reply:
As @candita already mentioned. The only error from `Run()` is when junit fails. If a conformance run fails some of the tests then `err` will be `nil`.
 

```console
$ ./ingress-operator h2spec -t -h www.redhat.com
Finished in 13.2878 seconds
146 tests, 137 passed, 4 skipped, 5 failed
2022-05-03T11:51:55.190+0100	ERROR	operator.h2spec	h2spec/client.go:29	error
```

Reply:
On the openshift/origin side we assert success or failure according to the comments here: https://github.com/openshift/origin/blob/master/test/extended/router/h2spec.go#L247